src/main.c

```diff
@@ -214,7 +212,50 @@ static void set_sys_messages(ConfigManager *configmgr) {
   zl_context.trim_sys_message = trim;
 }
 
+//size of "ZWE_zowe_sysMessages"
+#define ZWE_ZOWE_SYS_MESSAGES "ZWE_zowe_sysMessages"
+#define ZWE_ZOWE_SYS_MESSAGES_LEN (sizeof(ZWE_ZOWE_SYS_MESSAGES) - 1)
+
+static bool check_match_and_wto_message(const char* sys_message_id, const char* input_string, const bool other_messages) {
+
+  char *sys_message_start = strstr(input_string, sys_message_id);
+  int sys_message_pos = (sys_message_start != NULL) ? (sys_message_start - input_string) : -1;
+  if (sys_message_pos == -1) {
+    return false;
+  }
+  int input_string_len = strlen(input_string);
+
+  if (other_messages) {
+    // App-server: "Show Environment"
+    // E.g.ZWE_zowe_sysMessages_0=ZWEL0021I
+    if (memcmp(ZWE_ZOWE_SYS_MESSAGES, input_string, ZWE_ZOWE_SYS_MESSAGES_LEN) == 0) {
+      return 0;
+    }
+    // App-server: ZWED5015I prints config as json
+    // All zowe.sysMessages are printed, this is trying to ignore it
+    // Eg: |     "ZWED0031I",| -> strlen("ZWED0031I") + 9 extra characters
+    if (input_string_len <= strlen(sys_message_id) + 9) {
```

Comment:
A magic number 9 :) i see where it comes from, but probably we can define it above and describe it we will keep this solution option.